kernel/arch/dreamcast/hardware/syscalls.c

```diff
@@ -0,0 +1,235 @@
+/* KallistiOS ##version##
+
+   syscalls.c
+
+   Copyright (C) 2024 Andy Barajas
+*/
+
+#include <stdio.h>
+
+#include <dc/syscalls.h>
+#include <arch/memory.h>
+
+/*
+    This module along with syscall_font.s contains functions for accessing 
+    the Dreamcast's system calls.
+*/
+
+/*  
+    Each system call is performed via an indirect vector (VEC_*). This means 
+    that rather than calling a fixed address, a function pointer is fetched 
+    from the fixed address and the call goes through this pointer.
+*/
+#define VEC_SYSINFO       (MEM_AREA_P1_BASE | 0x0C0000B0)
+#define VEC_BIOFONT       (MEM_AREA_P1_BASE | 0x0C0000B4)
+#define VEC_FLASHROM      (MEM_AREA_P1_BASE | 0x0C0000B8)
+#define VEC_MISC_GDROM    (MEM_AREA_P1_BASE | 0x0C0000BC)
```

Comment:
BTW here exists additional syscall vector for GD-ROM - 0x0C0000C0
https://github.com/DC-SWAT/DreamShell/blob/master/firmware/isoldr/loader/gdc_syscall.s#L193

Reply:
Here is the assembly from your link.  Can you give me a little bit more detail?  What does it do?  What params does it take if any?
`gdc_redir_c0:
	mov     r7, r0
	mov     #16, r1
	cmp/hi  r0, r1
	bf      bad_syscall
	mov.l   gdc_syscall, r1
	shll2   r0
	mov.l   @(r0, r1), r0
	jmp     @r0
	nop`

Reply:
I need to look into it tomorrow but I see that you first check if it's a misc function first. Don't understand the middle potion yet but you return 0 on success and -1 on failure

Reply:
Here I just simulate BIOS syscalls code, I don't know for what this checking, maybe for dev/test purposes. Maybe I knew, but I don’t remember.
It can be useful in some cases. If used ISO redirection in dc-load for example, you can still use original GD syscalls by second vector.

Reply:
@DC-SWAT I added ```#define VEC_GDROM2        (MEM_AREA_P1_BASE | 0x0C0000C0)``` but still am unsure how to apply it.  I looked around at some emulators (redream/flycast) and it looks like VEC_MISC_GDROM and VEC_GDROM2 are interchangable when it comes to executing GDROM syscalls.  If you have any more tips or advice on what to do with VEC_GDROM let me know.  Otherwise it could be easily added later I think.

Reply:
I know of only one case when this can be used and I have already described it. I had another one in the loaders once case, but now it is no longer relevant, because iso loader has custom ATAPI driver for testing on emulators.
Just let it be here to know that it exists. Maybe it will be useful to someone someday.